Flutter_TDD_Architecture_Course_notes.md

```diff
@@ -1 +1,271 @@
+All the features of the application will be divided into three folders, **domain**, **data** and **presentation**.
+It is also explained how each of these folders works and what they will contain.
+
+<br />
+
+#### 2. Entities & Use Cases 
+> Video:https://www.youtube.com/watch?v=lPkWX8xFthE&list=PLB6lc7nQ1n4iYGE_khpXRdJkJEp9WOech&index=2
+
+In the `second tutorial` we start by adding the packages of the dependencies that will be needed to build the application.
+
+````
+dependencies:
+  flutter:
+    sdk: flutter
+
+  get_it: ^2.0.1
+  
+  # Bloc for state management
+  flutter_bloc: ^0.21.0
+  
+  # Value equality
+  equatable: ^0.4.0
+  
+  # Functional programming thingies
+  dartz: ^0.8.6
+  
+  # Remote API
+  connectivity: ^0.4.3+7
+  
+  http: ^0.12.0+2
+  # Local cache
+  shared_preferences: ^0.5.3+4
+
+  cupertino_icons: ^0.1.2
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  mockito: ^4.1.0
+  ````
+  
+  It is explained how the **entity** will work in this application.
+  The following explains the function of the **use cases** that will collect data from the repositories.
+  The use cases function in this application will collect the entities and show them in the presentation folder where all the elements that we will use physically in our application are located.
+  
+  
+  #### The either Type
+  
+  The **either** type is shown and explained with the function of being able to represent two types at the same time in this case serves for the errors and for the numbers that we will show.
+In which **L** are the **errors** and **R** the **numbers**.
+
+``Future<Either<Failure, NumberTrivia>>``
+ 
+
+Then it is defined the **failures** that will have an own file.
+
+
+
+> Failures.dart
+
+````
+import 'package:equatable/equatable.dart';
+
+abstract class Failure extends Equatable{
+  Failure([List properties = const<dynamic>[]]): super (properties);
+}
+
+````
+
+Next, define content for the `number_trivia_repository.dart` file.
+
+
+````
+import 'package:clean_architeture_tdd/core/error/failures.dart';
+import 'package:clean_architeture_tdd/features/number_trivia/domain/entities/number_trivia.dart';
+import 'package:dartz/dartz.dart';
+
+abstract class NumberTriviaRepository{
+  Future<Either<Failure, NumberTrivia>> getConcreteNumberTrivia(int number);
+  Future<Either<Failure, NumberTrivia>> getRandomNumberTrivia();
+}
+
+````
+
+Add the file `get_concrete_number_trivia.dart` to the `usecases` folder.
+
+![get_concrete_number_trivia_file](https://user-images.githubusercontent.com/27420533/74527004-92469800-4f1c-11ea-8f14-d02355c219fe.png)
+
+
+Then the **first** test will be created.
+
+Inside the `test` folder and create another file with the name `get_concrete_number_trivia_test.dart`.
+
+
+![get_concrete_number_trivia_test](https://user-images.githubusercontent.com/27420533/74527183-f5382f00-4f1c-11ea-9f4f-05d501a20bab.png)
+
+
+This will be the code for the **first** test.
+
+
+````
```

Comment:
@miguelmartins17 _four_ backtick characters is not supported by Markdown. 
_Please_ read: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code

If you used _three_ backticks and the keyword `dart` the syntax highlighting will work.

Reply:
I tried to use less before but it didn't appear to me as a code appeared to be a normal sentence.